modules/administration-guide/partials/proc_configuring-minikube-github-authentication.adoc

```diff
@@ -0,0 +1,59 @@
+[id="configuring-minikube-github-authentication_{context}"]
+= Configuring Minikube with GitHub Authentication
+
+== Authentication on Kubernetes
+On {kubernetes}, link:https://github.com/oauth2-proxy/oauth2-proxy[OAuth2 Proxy] is deployed as che-gateway sidecar ensuring requests authentication. Then {kubernetes} must have configured OIDC issuer to know about users identities and CheCluster CR must be configured with:
+```
+spec:
+  auth:
+    identityProviderURL: <oidc-issuer-url>
+    oAuthClientName: <oauth-client-id>
+    oAuthSecret: <oauth-client-secret>
+```
+
+Chectl can configure Minikube with link:https://dexidp.io/[Dex] as OIDC issuer with preconfigured static users. This is the default setup when deploying on Minikube with {devworkspace} engine `chectl server:deploy --platform minikube --workspace-engine=dev-workspace`. Dex can serve as a bridge to 3rd party {identity-provider}, like GitHub.
+
+=== Setup GitHub as identity provider on Kubernetes
+To setup GitHub as {prod-short} identity provider
+
+. Create OAuth App in GitHub at https://github.com/settings/applications/new (see link:https://docs.github.com/en/developers/apps/building-oauth-apps/creating-an-oauth-app[GitHub documentation]):
++
+[source]
+----
+Application name: Eclipse Che <1>
+Homepage URL: https://<minikube_ip>.nip.io <2>
+Authorization callback URL: https://dex.<minikube_ip>.nip.io/callback <3>
+----
++
+<1> Name is only displayed on GitHub. It is not used internally so it can be any name.
+<2> Main URL to Che instance.
+<3> Callback URL to Dex. Chectl deploys Dex on `dex.` subdomain.
++
+Note: To get Minikube IP, run `$ minikube ip` in the terminal.
+
+
+. On GitHub Generate new client secret for just created OAuth application
+
+. edit Dex configmap `kubectl edit configmap dex -n dex`
```

Comment:
To be consistent with the other docs, shouldn't we rather edit the CheCluster CR?

Reply:
dex configuration is not configured by CheCluster CR

Reply:
You mean it is an additional configuration that comes on top of "configure OIDC issuer"?

> Then {kubernetes} must have configured OIDC issuer to know about users identities and CheCluster CR must be configured with:

```
spec:
  auth:
    identityProviderURL: <oidc-issuer-url>
    oAuthClientName: <oauth-client-id>
    oAuthSecret: <oauth-client-secret>
```

Reply:
I don't understand how creating a CheCluster CR results in a Dex deployment without configuration.

Reply:
Do you mean something like: 

. Creating a CheCluster CR results in a Dex deployment with a default configuration with static users.

. To customize the Dex configuration, manual steps are required ...

Reply:
No. Dex is not configured by CheCluster CR at all. Chectl is the one who deploys and configures Dex and minikube to use Dex as OIDC provider. The configuration in CheCluster CR is just configuring Che to use the Dex instance. In case of different kubernetes flavors (AWS, Google, ...) we will require to set them with OIDC provider and set the same values in CheCluster CR to connect to OIDC provider of their choice.